src/libraries/System.Console/tests/ConsoleEncoding.Windows.cs

```diff
@@ -38,13 +41,38 @@ public void InputEncoding_SetUnicodeEncoding_SilentlyIgnoredInternally()
         }).Dispose();
     }
 
+    [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public void InputEncoding_SetEncodingWhenDetached_ErrorIsSilentlyIgnored()
+    {
+        RemoteExecutor.Invoke(() =>
+        {
+            Encoding encoding = Console.InputEncoding.CodePage != Encoding.ASCII.CodePage
+                ? Encoding.ASCII
+                : Encoding.Latin1;
+
+            // use FreeConsole to detach the current console - simulating a process started with the "DETACHED_PROCESS" flag
+            FreeConsole();
+
+            // Setting the input encoding should not throw an exception
+            Console.InputEncoding = encoding;
+            // The internal state of Console should have updated, despite the failure to change the console's input encoding
+            Assert.Equal(encoding, Console.InputEncoding);
+            // Operations on the console are no longer valid - GetConsoleCP fails.
+            Assert.Equal(0u, GetConsoleCP());
+        }).Dispose();
+    }
+
     [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
     [PlatformSpecific(TestPlatforms.Windows)]
     public void OutputEncoding_SetDefaultEncoding_Success()
     {
         RemoteExecutor.Invoke(() =>
         {
-            Encoding encoding = Encoding.GetEncoding(0);
+            Encoding encoding = Console.OutputEncoding.CodePage != Encoding.ASCII.CodePage
```

Comment:
This test seems to be passing 0 specifically to get the system's default encoding, even if it's the current encoding